### PR TITLE
Handle processing reminders for calendar objects with an infinite number of recurrences

### LINKED ISSUE #29896: Calendar reminders with infinite recurrence
Steps to reproduce:
* Create an event with recurrence options and **no end condition**. Add any kind of reminders to it. Save it.
* Run the process reminder CLI task `dav:send-event-reminders`

```
In EventIterator.php line 299:
                                                      
  Recurring events are only allowed to generate 3500  
                                                      

dav:send-event-reminders
```

We should catch `MaxInstancesExceededException` and ignore so that it doesn't block the rest of the processing.

### PR DESCRIPTION
**Pull request by tcitworld**
Closes #29896

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]

File: apps/dav/lib/CalDAV/Reminder/ReminderService.php [hunk 1]
```diff
@@ -247,6 +248,10 @@ public function onCalendarObjectCreate(array $objectData):void {
 				// instance. We are skipping this event from the output
 				// entirely.
 				return;
+			} catch (MaxInstancesExceededException $e) {
+				// The event has more than 3500 recurring-instances
```
Comment (miaulalala) [inline apps/dav/lib/CalDAV/Reminder/ReminderService.php:252]: add some logging for this exception?

**Review by miaulalala — APPROVED** [review]
LGTM

File: apps/dav/lib/CalDAV/Reminder/ReminderService.php (hunk 1 above)
Comment (tcitworld) [reply apps/dav/lib/CalDAV/Reminder/ReminderService.php:252]: I mean it's not a case where the admin can do anything about it anyway, so I was lazy and didn't want to inject ILogger just for this.

Comment (miaulalala) [reply apps/dav/lib/CalDAV/Reminder/ReminderService.php:252]: Understandable, have a good day ;)

**Pull request merged**